datasource/myharvard.go

```diff
@@ -40,9 +43,17 @@ func mhSearchRaw(search map[string]any) ([]any, error) {
 		return nil, fmt.Errorf("post request had bad status code: %v", resp.Status)
 	}
 
+	// Store the beginning of the response body into a buffer, so we can log it
+	// if there's an error.
+	var msgBuilder strings.Builder
+	msgBuilder.Grow(1024)
+	msgWriter := limitio.NewWriter(&msgBuilder, msgBuilder.Cap(), true)
```

Comment:
This is correct but probably too complicated, just use `io.ReadAll()` into a string and `json.Unmarshal()` instead of the combined `json.NewDecoder().decode()` function.

The decoder reads the entire string into memory first anyway for efficiency reasons.

Reply:
Ok! It feels more natural to me to use a decoder on a stream of data, and the current way will remain safe if the internals of the decoder change and my.harvard decides to drop a 1 GB HTML file on us (however unlikely).

Still, this is just a manual script, doing it the simpler way does make the code cleaner and more maintainable. I'll change it! We can always go back.

Reply:
It's fine if you want to do it this way, too! I don't want to nitpick, feel free to use limitio, not the most critical piece of code. Just wanted to make sure we're informed, [the JSON decoder internals won't change](https://github.com/golang/go/issues/11046#issuecomment-209646316)

Reply:
Cool, thanks for the link!